### DISCUSSION TITLE
Add a stack of equipment

### DISCUSSION THREAD

**Discussion by TrixaFett** [netbox-community/netbox#10614] [Ideas]
Create a stack where you add a device type and then +1 to the stack all will be added underneath as a device with s/n and relevant information. Also each interface  it could also add 1 pr stack x/0/1 where x is the number in stack.

We are using stacked switched, they share name and share IP and configuration, all of which can be connected against.
as i see it the only way is to create a stack, and then create individual devices, but they share the same name, and cannot create multiple devices with same name.

I was searching for this topic and found some that was closed, but still do not fix this:
Support for Switch Stack #940
Allow creation of a stack of devices #99
Release v2.3.0 #1913

**Comment by candlerb** [comment]
This feature already exists, it is called "Virtual chassis".

The only thing you need to beware of is to rename the interfaces on the 2nd and subsequent devices, before joining them to virtual chassis.  There is an interface bulk rename function for that.

For example: if the device type has interfaces `Gi0/[1-24]`, then after instantiating the second device you'll want to rename them to `Gi1/[1-24]` before joining it to the virtual chassis.

**Reply by cyberndj** [reply]
> after instantiating the second device you'll want to rename them to `Gi1/[1-24]` before joining it to the virtual chassis.

^^ Will help ALOT when your looking at all the interfaces. Its annoying seeing 3-8 instances of the same named interface...

Also, don't forget to set the virtual chassis master.
